VikingQuiz.UI/src/Components/MyQuizesPage/DisplayPlayedGamesComponent/DisplayPlayedGamesComponent.tsx

```diff
@@ -0,0 +1,144 @@
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            games: [],
+            gameData: [],
+            noGamesPlayed: true,
+            sortedGameDataByDate: []
+        }
+
+        this.SetupHandlers();
+    }
+
+    public SetupHandlers = () => {
+        this.updatePlayedGamesListOnQuizDelete = this.updatePlayedGamesListOnQuizDelete.bind(this);
+        this.componentWillMount = this.componentWillMount.bind(this);
+        this.sortPlayedGamesByDate = this.sortPlayedGamesByDate.bind(this);
+        this.getPlayedGames = this.getPlayedGames.bind(this)
+    }
+
+    public componentWillMount(){   
+        this.getPlayedGames()
+        this.sortPlayedGamesByDate()
+    }
+
+    public getPlayedGames(){
+        this.httpService.getWithToken(this.apiAddressForGames)
+        .then((response: any) => {
+            this.setState({ games: response.data }) 
+            response.data.forEach((game: any) => {
+                let gameNumberOfPlayers:any;
+                let quizDetails: any;                   
+                this.httpService.getWithToken(this.apiAdressForPlayerGame + game.id) 
+                    .then((result: any) => {
+                        gameNumberOfPlayers = result.data   
+                        const gameDataCopy = this.state.gameData;
+                        this.httpService.getWithToken(this.apiAddressForQuizzes + game.quizId)
+                            .then((res: any) => {
+                                quizDetails = res.data                                                 
+                                gameDataCopy.push({
+                                    numberOfPlayers: gameNumberOfPlayers,
+                                    gameDate: game.gameDate,
+                                    quizTitle: quizDetails.title,
+                                    quizPhoto: quizDetails.pictureUrl
+                                    })     
+                                    this.setState({
+                                        gameData: gameDataCopy,
+                                        noGamesPlayed: false
+                                    })     
+                                    this.sortPlayedGamesByDate();                                                                                          
+                            });  
+                    })
+            })
+        })
+        .catch((error: any) => console.log(error))  
+    }
+
+    public sortPlayedGamesByDate()
+    {
+        this.setState({
+            sortedGameDataByDate : this.state.gameData
+                .sort((a: any, b: any) => a.gameDate.split('/').join()
+                .localeCompare(b.gameDate.split('/').join()))
+        })
+    }
+       
+    public updatePlayedGamesListOnQuizDelete(){
+        this.setState({
+            gameData: [],
+            noGamesPlayed: true 
+        })
+        this.componentWillMount()
+    }
+
+
+    public render(){     
```

Comment:
A bit too much JSX code. You should try to divide it into small components to make it more condensed.